### PR TITLE
Activate/toggle buttons with space on keyup

### LINKED ISSUE #610: Button: Space should only activate on keyup
The synthetic click event that happens on native `<button>` elements occurs on keydown for Enter and keyup for Space, but the button pattern doesn't describe this behavior and the example does not exhibit it. For instance, holding down the space bar on the "Mute" button toggles it off and on until you release the space bar--it should only trigger once on key release.

I tested this behavior using [this codepen example](https://codepen.io/sh0ji/pen/vdXraZ) on Firefox, Chrome, and Safari on Mac to be certain about it. Holding down space bar on the "real button" only triggers the click even on release, whereas holding down enter triggers it repeatedly until release. I added a "fake button" to demonstrate how this behavior can be replicated in JavaScript.

### PR DESCRIPTION
**Pull request by kleinfreund**
Implements #610.

Both action and toggle buttons are now activated with space on the keyup event and no longer on the keydown event. The behavior for pressing enter has not changed.

As brought up in #610, I would like to remove the safety check on lines 66 to 69 (in the new file). What are your thoughts on that?

Also, do you want me to expand the description in the design pattern to reflect this behavior?

### CODE REVIEW

File: examples/button/js/button.js [hunk 1]
```diff
@@ -2,84 +2,114 @@
...
+}
+
+/**
+ * Activates the action button with the enter or space key.
+ *
+ * @param {KeyboardEvent} event
+ */
+function actionButtonKeyupHandler (event) {
+  // If either enter or space is pressed, activate the button
+  if (event.keyCode === 13 || event.keyCode === 32) {
```
Comment (sh0ji) [inline examples/button/js/button.js]: The <kbd>Enter</kbd> check should be removed here. It's causing `activateActionButton` to trigger twice on <kbd>Enter</kbd> keypresses, effectively making the toggle button pointless.

File: examples/button/js/button.js [hunk 2]
```diff
@@ -2,84 +2,114 @@
...
-  var button = event.target;
-  var currentState = button.getAttribute('aria-pressed');
-  var newState = 'true';
+/**
+ * Toggles the toggle button’s state with the enter or space key.
+ *
+ * @param {KeyboardEvent} event
+ */
+function toggleButtonKeyupHandler (event) {
+  if (event.keyCode === 13 || event.keyCode === 32) {
```
Comment (sh0ji) [inline examples/button/js/button.js]: Same thing as line 49 here.

File: examples/button/js/button.js [hunk 3]
```diff
@@ -2,84 +2,114 @@
...
-    newState = 'false';
-    newIconState = ICON_SOUND_URL;
-  }
+  button.setAttribute('aria-pressed', isAriaPressed ? 'false' : 'true');
 
-  // Set the new aria-pressed state on the button
-  button.setAttribute('aria-pressed', newState);
-  icon.setAttribute('xlink:href', newIconState);
+  var icon = button.querySelector('use');
+  icon.setAttribute('xlink:href', isAriaPressed ? ICON_SOUND_URL : ICON_MUTE_URL);
```
Comment (sh0ji) [inline examples/button/js/button.js]: 👍

File: examples/button/js/button.js (hunk 1 above)
Comment (kleinfreund) [reply examples/button/js/button.js]: Oh, what a blunder. Of course.

**Comment by kleinfreund** [comment]
I updated the pull request according to @sh0ji’s review.

**Comment by kleinfreund** [comment]
Not sure if that Travis CI check is supposed to complete. It seems like it got stuck?

**Comment by caasi** [comment]
My friend is working on her own ARIA practices. We are wondering if there is any reference about this keyup behavior other than the "Change History" section of [WAI-ARIA Authoring Practices 1.1](https://www.w3.org/TR/wai-aria-practices-1.1/#change_log_2019_january). Can anyone point a direction?

**Comment by caasi** [comment]
> Is it open source?

Currently it's just a [code sandbox](https://twitter.com/miau715/status/1184858504135245824).

> This was done to better match what the HTML button element does on native platforms.

You are right. Another friend @othree found an answer about this behavior here: https://stackoverflow.com/questions/16090578/why-do-enter-and-space-keys-behave-differently-for-buttons

**Pull request merged**